test/e2e/metallb_test.go

```diff
@@ -0,0 +1,279 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/aws/eks-anywhere/pkg/curatedpackages"
+	"github.com/aws/eks-anywhere/test/framework"
+	"github.com/stretchr/testify/suite"
+)
+
+type Suite struct {
+	suite.Suite
+	cluster *framework.ClusterE2ETest
+}
+
+func TestPackagesMetalLB(t *testing.T) {
+	suite.Run(t, new(Suite))
+}
+
+func (suite *Suite) SetupSuite() {
+	t := suite.T()
+	suite.cluster = framework.NewClusterE2ETest(t, framework.NewDocker(t),
+		framework.WithPackageConfig(t, EksaPackageBundleURI, EksaPackageControllerHelmChartName,
+			EksaPackageControllerHelmURI, EksaPackageControllerHelmVersion,
+			EksaPackageControllerHelmValues),
+	)
+}
+
+func (suite *Suite) TestPackagesMetalLB() {
+	// This should be split into multiple tests with a cluster setup in `SetupSuite`.
+	// This however requires the creation of utilites managing cluster creation.
+	t := suite.T()
+	suite.cluster.WithCluster(func(test *framework.ClusterE2ETest) {
+		test.InstallCuratedPackagesController()
+		ctx := context.Background()
+		kubeconfig := test.Cluster().KubeconfigFile
+		namespace := "metallb-system"
+		t.Cleanup(func() { test.DeleteNamespace(namespace) })
+		test.CreateNamespace(namespace)
+		packageName := "metallb"
+		packagePrefix := "test"
+		err := WaitForLatestBundleToBeAvailable(test, ctx, 2*time.Minute)
+		if err != nil {
+			t.Fatalf("waiting for latest bundle: %s", err)
+		}
+		bundle, err := GetLatestBundleFromCluster(test)
+		if err != nil {
+			t.Fatal(err)
+		}
+		err = curatedpackages.UpgradePackages(ctx, kubeconfig, test.KubectlClient, curatedpackages.Cluster, bundle)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		t.Run("Basic installation", func(t *testing.T) {
+			t.Cleanup(func() { test.UninstallCuratedPackage(packagePrefix) })
+			test.InstallCuratedPackage(packageName, packagePrefix, "--kube-version=1.22")
+			err = WaitForPackageToBeInstalled(test, ctx, packagePrefix, 30*time.Second)
+			if err != nil {
+				t.Fatalf("waiting for metallb package to be installed: %s", err)
+			}
+			err = test.KubectlClient.WaitForDeployment(context.Background(),
+				test.Cluster(), "1m", "Available", "test-metallb-controller", namespace)
+			if err != nil {
+				t.Fatalf("waiting for metallb controller deployment to be available: %s", err)
+			}
+			err = WaitForDaemonset(test, ctx, "test-metallb-speaker", namespace, 2, 30*time.Second)
```

Comment:
Nitpick: The "2" here is a magic number. It's in no way clear what it's purpose is without knowing the function arguments, and even knowing what it is, it's not clear why we're waiting for 2 nodes?

Reply:
The 30 seconds at least we can imply from the function name.